plugins-extra/query_enhancements/public/query_assist/components/query_assist_input.tsx

```diff
@@ -0,0 +1,75 @@
+import { EuiFieldText, EuiIcon, EuiOutsideClickDetector, EuiPortal } from '@elastic/eui';
+import React, { useMemo, useState } from 'react';
+import {
+  PersistedLog,
+  QuerySuggestionTypes,
+  SuggestionsComponent,
+} from '../../../../../src/plugins/data/public';
+import assistantLogo from '../../assets/query_assist_logo.svg';
+
+interface QueryAssistInputProps {
+  inputRef: React.RefObject<HTMLInputElement>;
+  persistedLog: PersistedLog;
+  initialValue?: string;
+  selectedIndex?: string;
+  previousQuestion?: string;
+}
+
+export const QueryAssistInput: React.FC<QueryAssistInputProps> = (props) => {
+  const [isSuggestionsVisible, setIsSuggestionsVisible] = useState(false);
+  const [suggestionIndex, setSuggestionIndex] = useState<number | null>(null);
+  const [value, setValue] = useState(props.initialValue ?? '');
+
+  const recentSearchSuggestions = useMemo(() => {
+    if (!props.persistedLog) return [];
+    return props.persistedLog
+      .get()
+      .filter((recentSearch) => recentSearch.includes(value))
+      .map((recentSearch) => ({
+        type: QuerySuggestionTypes.RecentSearch,
+        text: recentSearch,
+        start: 0,
+        end: value.length,
+      }));
+  }, [props.persistedLog, value]);
+
+  return (
+    <EuiOutsideClickDetector onOutsideClick={() => setIsSuggestionsVisible(false)}>
```

Comment:
should we check with ux to see if they want us to also submit the question if set on an outside click?

Reply:
then if user clicks the data-grid to copy a field, since it's an outside click, their incomplete question will be submitted. i think that can be weird?